### ISSUE TITLE
Is it possible define r-value of construction as a decision variable?

### ISSUE THREAD

**Issue by orcunkoraliseri** [santoshphilip/eppy#190]
In eppy library, as i checked r-value method is available both materials and constructions. I am currently
working on a project to how sensitive the r-values of building materials in terms of energy demand of unit.
Related with it, I also tried to arrange r-values for envelope constructions. But, I could not.

Is it possible? And if it is possible how can i do it?

**Comment by santoshphilip** [comment]
What do you mean by  arrange, "I also tried to arrange r-values for envelope constructions". Did you mean sort it ? Do you have a code snippet you can post ?

**Comment by orcunkoraliseri** [comment]
Actually, I don't have any code snippet, I just think if it is possible?

**Comment by santoshphilip** [comment]
@orcunkoraliseri , sorry about the delayed response. For some reason I am not getting notification on this post. So I just saw it.

I am not understanding what you are trying to do. Can you explain in a different way, or give an example

**Comment by orcunkoraliseri** [comment]
Sorry about mistelling. I am currently working on a project for early design architectural energy modelling. Related with this, I prepared a sensitivity analysis framework to analyse decision variables (building height, orientation, u-value of glazing, etc.) influence on total energy demand. I also have intention to analyse for building envelope for r-value of materials (brick, concrete, etc.) in the current framework. Instead of define to r-value of materials as decision variables, I thought maybe it is more suitable to search on r-value of construction with its own layers at early design energy modelling. 

I checked eppy thermal properties and used to observe construction or material r-value, but it gives the r-value of construction, but i am intended to change r-value of construction. 

`CONSTRUCTION,
    Exterior Roof,            !- Name
    M14a 100mm heavyweight concrete,    !- Outside Layer
    F05 Ceiling air space resistance,    !- Layer 2
    F16 Acoustic tile;        !- Layer 3`


Is it possible to set r-value of construction as a decision variable? I mean, is it possible to change r-value of construction by using r-value method of eppy or possible to change all layers separately?

**Comment by santoshphilip** [comment]
OK ... I got what you are trying to do.
I ran into a similar problem while writing code for [py90dot1](https://github.com/pyenergyplus/py90dot1). The model needed  constructions that had a particular r-value as defined in the ASHRAE tables. One way of doing this was to use a `Material:NoMass` in the construction. See an example of this [here](https://github.com/pyenergyplus/py90dot1/blob/master/py90dot1/ASHRAE_constr/ASHRAE_constr.py)

Note that if you do this your construction won't have the thermal mass properties that brick or concrete would have. But it would be still good for a first pass on a parametric run. 

If you want to include thermal mass in your modeling, put in the other materials like brick concrete wood etc and also put in `Material:NoMass`. Change  the variable `Resistance` in `Material:NoMass` to gradually change the r-value.
